### ISSUE TITLE
discrepancy between species counts

### ISSUE THREAD

**Issue by kueda** [inaturalist/inaturalist#1370]
Maybe I'm not understanding something, but there's a discrepancy between the species counts at these two links:

http://www.inaturalist.org/observations/stats?place_id=35&taxon_id=47126&d1=2017-01-01&d2=2017-12-31&quality_grade=research
http://www.inaturalist.org/observations?d1=2017-01-01&d2=2017-12-31&place_id=35&quality_grade=research&taxon_id=47126

**Comment by loarie** [comment]
Thats driven by how species are counted. The latter is counting rolling ssp up to species and including leaves. I believe the former is just counting species

This brings them closer:
http://www.inaturalist.org/observations/stats?place_id=35&taxon_id=47126&d1=2017-01-01&d2=2017-12-31&quality_grade=research&rank=species

http://www.inaturalist.org/observations?d1=2017-01-01&d2=2017-12-31&hrank=species&lrank=species&place_id=35&quality_grade=research&taxon_id=47126&view=species

but they still differ by 2 obs - not sure why

**Comment by kueda** [comment]
Yeah, and there's a 3 obs discrepancy between these two as well (I think that's actually what I originally meant to post):

http://www.inaturalist.org/observations/stats?place_id=35&taxon_id=47126&d1=2017-01-01&d2=2017-12-31&quality_grade=research&rank=leaves
http://www.inaturalist.org/observations?d1=2017-01-01&d2=2017-12-31&place_id=35&quality_grade=research&taxon_id=47126

**Comment by boucur** [comment]
Scott is right - the Observations page only counts species. It doesn't count subspecies or lower taxa, but it should, right? Or at least it should display them on the species "tab"? (right now lower taxa are excluded from that page, e.g. neither Osmunda regalis var. spectabilis nor Osmunda regalis are shown because only Osmunda regalis var. spectabilis has been observed and it is not a straight species)

The two pages **do match** when the hrank and lrank are both set to "species."
http://www.inaturalist.org/observations/stats?place_id=35&taxon_id=47126&d1=2017-01-01&d2=2017-12-31&quality_grade=research&hrank=species&lrank=species
http://www.inaturalist.org/observations?d1=2017-01-01&d2=2017-12-31&place_id=35&quality_grade=research&taxon_id=47126&view=species&hrank=species&lrank=species

They **do not match** when hrank is species and lrank is subspecies:
http://www.inaturalist.org/observations/stats?place_id=35&taxon_id=47126&d1=2017-01-01&d2=2017-12-31&quality_grade=research&hrank=species&lrank=subspecies
http://www.inaturalist.org/observations?d1=2017-01-01&d2=2017-12-31&place_id=35&quality_grade=research&taxon_id=47126&view=species&hrank=species&lrank=subspecies

(I'm not sure how rank=species or rank=leaves function... or why those two pairs of links differ by so few spp.)

**Comment by pleary** [comment]
I don't think I'm seeing the problem. The counts of "species" are not calculated the same way on the older stats pages and the new obs search, as Scott said. Scott's two links were off because one was using `hrank`/`lrank` and the other was using `rank`. The species counts will match up if you use the same params in each place - `hrank=species&lrank=species` gives a species count of 614 in both places, and `rank=species` gives a count of 612 in both places. The reason these are different is there are clearly some obs with `rank_level=10` that are not species (probably hybrids).

It's also not the case the the old "leaves" is the same as how we count "species" in the new obs search. They are different calculations that give different numbers, and leaves will, I think, always be the same or larger than the "species" counts on obs search.

We can definitely start to sync things up, but I think the idea was to roll out the new stuff w/o changing the behavior of the old stuff. Different numbers are to be expected since we're not being consistent about what "species" are.

@boucur the species tab of obs seach shouldn't really count species or subspecies, and it should never show anything below the species level. Rather it should look at the ancestries of all taxa in the results, cut off the ancestries at the species level, and count the distinct species represented by the results. If for example a search returned Aus bus, Aus bus cus, and Aus bus dus - that will be a species count of 1 (the species counted is Aus bus). If for example a search returned just Aus bus cus, that would also be a species count of 1 (the species is again Aus bus). This is different from the old concept of "leaves" which would return a count of 2 of the first example (counting the two subspecies) and a count of 1 in the second example.
